rust/gitxetcore/src/git_integration/git_wrap.rs

```diff
@@ -353,3 +355,181 @@ pub fn perform_git_version_check() -> Result<()> {
         }
     }
 }
+
+// Add files to a repo
+pub fn create_commit(
+    repo: &Arc<Repository>,
+    branch_name: Option<&str>,
+    commit_message: &str,
+    files: &[(&str, &[u8])],
+) -> Result<()> {
+    // Create blobs for the files
+    let file_oids: Vec<_> = files
+        .iter()
+        .map(|(file_name, data)| {
+            let blob_oid = repo.blob(data)?;
+            Ok((
+                file_name.to_owned().trim_end_matches('/'),
+                blob_oid,
+                data.len(),
+            ))
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+    // Create a tree from the blobs; do this in memory, then do it one by one.
+    let mut index = git2::Index::new()?;
+
+    for (file_name, file_oid, data_len) in file_oids {
+        let entry = git2::IndexEntry {
+            path: file_name.as_bytes().into(),
+            id: file_oid,
+            file_size: data_len as u32,
+            mode: 0o100644, // represents a blob (file)
+            dev: 0,
```

Comment:
From https://github.com/libgit2/libgit2/blob/45fd9ed7ae1a9b74b957ef4f337bc3c8b3df01b5/tests/libgit2/reset/hard.c#L105, these are fine to leave as zero.